### ISSUE TITLE
core support for connecting to remote UI via --server

### ISSUE THREAD

**Issue by xelra** [neovim/neovim#5773]
I want to be able to connect to a Neovim session on another computer. 

On the remote server I ran the following:
```
NVIM_LISTEN_ADDRESS=/tmp/nvim.socket nvim --headless
```
Then on the local machine I ran:
```
ssh -L /tmp/nvim.socket:/tmp/nvim.socket user@1.2.3.4
NVIM_LISTEN_ADDRESS=/tmp/nvim.socket nvim
```
Unfortunately that doesn't work though. Connecting to the socket via `NVIM_LISTEN_ADDRESS` is not possible. I've also tried it with `NVIM_LISTEN_ADDRESS=/tmp/nvim.socket nvim-qt` and `NVIM_LISTEN_ADDRESS=/tmp/nvim.socket nyaovim`.

Neovim-Qt has a `--server` option. Using that on my local machine `nvim-qt --server /tmp/nvim.socket` does work. So for now I can only connect to my remote nvim session with Neovim-Qt.

It would be great to see a `--server` option for the `nvim` binary too, so it's possible to also connect to remote Neovim sessions from the terminal.

**Comment by DoomHammer** [comment]
Hi! What's the current status of this? I have a Tmux plugin in mind that could use this.

I would like to have a global headless Neovim running all the time and be able to quickly open a new pane with Neovim UI connected to the headless instance. Tried some trickery with `reptyr`, but couldn't get any good results. Using `break-pane` and `join-pane` from tmux didn't help as well as it created a visilbe new window (I'd love it to be really background/transparent).

**Comment by bfredl** [comment]
abduco http://www.brain-dump.org/projects/abduco/ might be a more robust alternative to reptyr.

**Comment by DoomHammer** [comment]
Thanks for suggestion @bfredl !

**Comment by geekodour** [comment]
@bfredl is this a duplicate of https://github.com/neovim/neovim/issues/7438 ?

**Comment by bfredl** [comment]
well, this issue is older, so technically that one is duplicate of this :P

**Comment by rjshrjndrn** [comment]
Any update on this feature? This will be a lifesaver, if you're working with remote machines(lots of them).

**Comment by shaform** [comment]
Check out https://github.com/cdr/code-server, which allows remote connection to VSCode. And VSCode has Vim mode, so maybe it would be useful.

**Comment by rjshrjndrn** [comment]
@shaform Thank you for the suggestion. I checked out code-server. But that need docker in the remote server, and which won't be the case with all servers. And I don't actively use vs-code, as the vim emulation which it has can't have all mappings, plugins from my init.vim.

**Comment by justinmk** [comment]
related

- https://github.com/neovim/neovim/pull/18375
- https://github.com/neovim/neovim/pull/10071

**Comment by zeertzjq** [comment]
Closed by #18375
